client/src/Home.tsx

```diff
@@ -19,6 +20,10 @@ const mapStateToProps = (state: AppState, ownProps: InputProps) => ({
 class Home extends AppRouteComponent<typeof mapStateToProps> {
   componentDidMount() {
     this.props.thunkSetStateFromUrlBar();
+    history.listen((location, action) => {
```

Comment:
We probably need to unsubscribe from this somehow when the page is destructed.. This callback will forever be called even when we're not on Home anymore.

Reply:
Done.